libraries/app/api.cpp

```diff
@@ -302,26 +302,31 @@ namespace graphene { namespace app {
        const auto& db = *_app.chain_database();
        FC_ASSERT( limit <= 100 );
        vector<operation_history_object> result;
-       const auto& stats = account(db).statistics(db);
-       if( stats.most_recent_op == account_transaction_history_id_type() ) return result;
-       const account_transaction_history_object* node = &stats.most_recent_op(db);
+       account_statistics_object stats;
+       try { stats = account(db).statistics(db); } catch(...) { return result; }
+       const account_transaction_history_object* node;
+       try { node = &stats.most_recent_op(db); } catch(...) { return result; }
        if( start == operation_history_id_type() )
           start = node->operation_id;
+       if( start.instance.value > node->operation_id.instance.value )
+          start = node->operation_id;
 
-       while(node && node->operation_id.instance.value > stop.instance.value && result.size() < limit)
+       const auto& hist_idx = db.get_index_type<account_transaction_history_index>();
+       const auto& by_op_idx = hist_idx.indices().get<by_op>();
+       auto index_start = by_op_idx.begin();
+       auto itr = by_op_idx.lower_bound(boost::make_tuple(account, start));
+
+       while(itr != index_start && itr->operation_id.instance.value > stop.instance.value && result.size() < limit)
        {
-          if( node->operation_id.instance.value <= start.instance.value )
-             result.push_back( node->operation_id(db) );
-          if( node->next == account_transaction_history_id_type() )
-             node = nullptr;
-          else node = &node->next(db);
+          if(itr->account == account && itr->operation_id.instance.value <= start.instance.value)
+             result.push_back(itr->operation_id(db));
+          --itr;
+          if(itr->account != account) break;
```

Comment:
If you include itr->account == account in the while condition it will simplify the loop body.

Reply:
yes, this is a good one so we can remove the break. changed to it.